cdap-spark-core2_2.11/src/main/scala/co/cask/cdap/app/runtime/spark/BatchWritableFunc.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.runtime.spark
+
+import java.net.URI
+import java.util.concurrent.TimeUnit
+
+import co.cask.cdap.api.data.batch.BatchWritable
+import co.cask.cdap.api.dataset.Dataset
+import co.cask.cdap.data2.metadata.lineage.AccessType
+import org.apache.spark.TaskContext
+import org.apache.spark.broadcast.Broadcast
+import org.apache.tephra.TransactionAware
+
+import scala.collection.JavaConversions._
+
+/**
+  * In Spark2, the context output metrics are not available, so this version doesn't do anything with metrics
+  */
+object BatchWritableFunc {
+
+  /**
+    * Creates a function used by [[org.apache.spark.SparkContext]] runJob for writing data to a
+    * Dataset that implements [[co.cask.cdap.api.data.batch.BatchWritable]]. The returned function
+    * will be executed in executor nodes.
+    */
+  private[spark] def create[K, V](namespace: String,
+                                  datasetName: String,
+                                  arguments: Map[String, String],
+                                  txServiceBaseURI: Broadcast[URI]) = (context: TaskContext,
+                                                                       itor: Iterator[(K, V)]) => {
+
```

Comment:
this is like the Spark1 version except it doesn't use output metrics, which have been hidden in Spark2.